app/jobs/get_usps_proofing_results_job.rb

```diff
@@ -145,10 +149,16 @@ def handle_bad_request_error(err, enrollment)
   end
 
   def handle_standard_error(err, enrollment)
+    response_attributes = if err.respond_to?(:response)
+                            response_analytics_attributes(err.response)
```

Comment:
(Non-blocking but something to consider fixing up:)

The properties that we care about in `response_analytics_attributes` come from the `body` property of the response, not the response directly, though in the case of the error, I assume we won't have those properties anyways, so ultimately it makes no difference.

I'd suggest we either audit the responses we're passing to `response_analytics_attributes` to make sure we're always sending the response _body_, or (as in a prior comment) avoid calling `response_analytics_attributes` altogether in error paths where we're not expecting useful response properties to log.

Reply:
Actually, if we may be expecting `responseMessage` in the server error response (as implied by [some existing test cases](https://github.com/18F/identity-idp/blob/69ac7fd6fbde03cdf38127f60feebd1615c01e74/spec/services/usps_in_person_proofing/proofer_spec.rb#L159-L169)), we might actually want to fix this up.

```suggestion
                            response_analytics_attributes(err.response&.[](:body))
```

Reply:
Yeah I think there's some confusion stemming from our [Faraday middlewares](https://github.com/18F/identity-idp/blob/cf2bc30d251a8f08bbfaad9f722ddce886ba8a2e/app/services/usps_in_person_proofing/proofer.rb#L137-L143) which, for any <400 status code, will automatically parse the response body and return it as a hash. For any >=400 status code Faraday will raise an error and we'll have to explicitly access the body buried in the response (eg err.response[:body])

For example this is the response for a 2xx response status code:
```
{"status"=>"In-person passed",
 "proofingPostOffice"=>"WILKES BARRE",
 "proofingCity"=>"WILKES BARRE",
 "proofingState"=>"PA",
 "enrollmentCode"=>"2090002197504352",
 "primaryIdType"=>"Not supported",
 "transactionStartDateTime"=>"12/17/2020 033855",
 "transactionEndDateTime"=>"12/17/2020 034055",
 "fraudSuspected"=>false,
 "proofingConfirmationNumber"=>"350040248346701",
 "ippAssuranceLevel"=>"1.5"}
```

This is the err.response for a 4xx status code:
```
{:status=>400,
 :headers=>{},
 :body=>{"responseMessage"=>"More than 30 days have passed since opt-in to IPP"},
 :request=>
  {:method=>:post,
   :url_path=>"/ivs-ippaas-api/IPPRest/resources/rest/getProofingResults",
   :params=>nil,
   :headers=>{"Content-Type"=>"application/json; charset=utf-8", "User-Agent"=>"Faraday v1.8.0", "Authorization"=>"Bearer <redacted>", "RequestID"=>"d63e795a-e163-464b-9cd5-6b55e716ac7b"},
   :body=>"{\"sponsorID\":0,\"uniqueID\":\"<redacted>\",\"enrollmentCode\":\"<redacted>\"}"}}
```

I also noticed that there's a third type of behavior to account for. If we get a successful response (eg status code 2xx) but the JSON is invalid then Faraday will raise a Faraday::ParsingError. We can look at err.response to see the response object but its body will be nil. If we wanted to access the body (as unparsed text) in that case we could look at `err.response.env.response_body`. I think we should explicitly handle the `Faraday::ParsingError` as a new error case and handle it appropriately

Reply:
I'm going to create a new ticket to address auditing the different error cases and updating the code to make sure that it behaves as expected for the different types of responses we'll get back from Faraday for various error conditions

Reply:
Ticket to address this thread: https://cm-jira.usa.gov/browse/LG-7706